.github/workflows/github_ci_logs.yml

```diff
@@ -0,0 +1,32 @@
+name: Github CI log archiver
+on:
+  workflow_run:
+    workflows: ["Github CI"]
+    types:
+      - completed
+
+jobs:
+  save_logs:
+    name: Save run logs
+    runs-on: ubuntu-latest
+    steps:
+    - name: Download log archive
+      run: |
+        curl -sI -u ${{ github.repository_owner }}:${{ github.token }} \
+          https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.event.workflow_run.id }}/logs > response.txt
+        DOWNLOAD_LINK="https:"$(cat response.txt | grep -Fi Location: | cut -d: -f3)
+        DOWNLOAD_LINK=${DOWNLOAD_LINK%$'\r'}
```

Comment:
I had to look up what this line is doing. (`$'\r'` is a literal carriage return, and `${x%y}` trims a suffix, so this line is trimming any trailing carriage-return from the end of the `Location:` header extracted on the line above.)

I wonder if the above line could just be

    DOWNLOAD_LINK="https:$(cat response.txt | grep -Fi Location: | cut -d: -f3 | tr -d $'\r')"

or even

    DOWNLOAD_LINK="$(cat response.txt | grep -Fi Location: | cut -d: -f2 | tr -d $'\r')"

Is part of the intent that you need to change a literal `http` in the response to an `https` in the download link?

Reply:
> `$'\r'` is a literal carriage return, and `${x%y}` trims a suffix, so this line is trimming any trailing carriage-return from the end of the `Location:` header extracted on the line above.

Yep, had to add all that because `curl` didn't like the trailing CR (thanks, someone at StackOverflow).

> Is part of the intent that you need to change a literal `http` in the response to an `https` in the download link?

Nah, it's just that `grep -Fi Location:` produces a string that contains _two_ colons: "Location: https://{whatever}", so this is a lazy workaround with `cut`. I suppose something like `awk` would be a better fit for this.

Reply:
Ah, right, you have to change a leading `" https"` to `"https"`. This is the sort of problem I always want to use `grep -o` for, and it usually doesn't quite work, and I end up using `sed` instead. (I don't know `awk`, unfortunately.)

Does this work? (Slightly more hard-coded than yours. Although they'd be within their protocol rights to do so, I really hope Google isn't sending you a `Location:` header spelled like `LoCaTiOn:`. 😜)

    DOWNLOAD_LINK="$(cat response.txt | grep '^Location:' | grep -o 'https://\S*')"

Reply:
> I don't know `awk`, unfortunately.

That's what I thought, too, but it turned out much easier and elegant a solution (not that I know `awk` now, but hey):
```curl -JO $(awk -F ": " '{gsub("\r", "", $2); print $2}' response.txt)```
Also obviates your bash style nit above.